### PR TITLE
openapi2conv: fix that removes an extra `required` array from `formDataBody`

### PR DESCRIPTION
**Pull request by liankui**
issue #847

### CODE REVIEW

File: openapi2conv/issue847_test.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    }
+  }
+}
+`)
+	doc3, err := v2v3JSON(spec)
+	require.NoError(t, err)
+
+	reqStr, err := json.MarshalIndent(doc3, "", "    ")
+	require.NotNil(t, reqStr)
+	require.NoError(t, err)
```
Comment (fenollp) [inline openapi2conv/issue847_test.go:56]: This part seems irrelevant to the test.

File: openapi2conv/issue847_test.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
```
Comment (fenollp) [inline openapi2conv/issue847_test.go:57]: Please add both checks to `doc3`:
* one looking for the presence of the `required` array where you expect it
* one looking for the absence of the `required` array, you fixed

Comment (liankui) [reply openapi2conv/issue847_test.go:34]: I'm done, can you check it again?
